include/simdjson/padded_string.h

```diff
@@ -46,34 +50,37 @@ inline char *allocate_padded_buffer(size_t length) noexcept {
 // constructors.
 struct padded_string final {
 
+    using type = padded_string;
+    using value_type = char *;
   // free the previous payload
   // allocate for the new size
   // return false on ENOMEM
-  bool reset(size_t new_size_) noexcept 
+  static bool reset(type & pstring_, size_t new_size_) noexcept 
   {
-    this->~padded_string(); 
-    viable_size = new_size_;
-    data_ptr = allocate_padded_buffer(new_size_);
-    if (data_ptr != nullptr) {
-      data_ptr[new_size_] = '\0'; // easier when you need a c_str
+      if (pstring_.data())
+        pstring_.~type(); 
```

Comment:
I don't think calling the destructor on a live object is idiomatic.

Reply:
AFAK I am not doing it in this case. Advice: std::lib code is full of destructor calls.